General server dependency updates (was/including catalina fix)

dannon opened the pull request:
This started as an update to cryptography (the first commit here, which does fix startup on Catalina), but I decided that this early in the cycle would be a good time to do a full-pass update as well.  The first commit has been backported to 19.05 in #8845

dannon committed:
Cryptography update; this turns out to be the culprit causing python startups on catalina to fail (though I don't know exactly why)

dannon committed:
Version pinning with dev1, etc., like this doesn't work for >= vers

dannon committed:
Full dependency update

nsoranzo commented on lib/galaxy/dependencies/pipfiles/default/Pipfile:
```diff
@@ -70,7 +70,7 @@ bioblend = "*"
 kombu = "*"
 psutil = "*"
-pulsar-galaxy-lib = ">=0.14.0.dev1"
+pulsar-galaxy-lib = "==0.14.0.dev1"
```
This is necessary because of a pipenv issue: https://github.com/pypa/pipenv/issues/3928

nsoranzo committed:
Fix environment marker for `contextlib2`

- Avoid repeating the package name in sed regexp replacements.
- More dependency updates.

nsoranzo commented:
I just noticed that https://github.com/galaxyproject/galaxy/pull/8543 added 2 new dependencies to the Pipfile without running `make update-dependencies`: https://github.com/galaxyproject/galaxy/pull/8543/files#r338021231

This is now adding 9 new dependencies which I think may be not used: markdown, weasyprint, CairoSVG, Pyphen, cairocffi, cssselect2, html5lib, pdfrw and tinycss2.

nsoranzo committed:
Remove unused dependencies

Introduced in https://github.com/galaxyproject/galaxy/pull/8543 .

nsoranzo committed:
Fix environment marker for `pathlib2`
